dev-context/rest/rest-api-example-spec.yaml

```diff
@@ -0,0 +1,286 @@
+
+    RecipesCollectionItemV1:
+      description: Contains the recipe information.
+      oneOf:
+        - $ref: "#/components/schemas/RecipesCollectionDefaultItemV1"
+        - $ref: "#/components/schemas/RecipesCollectionDinnerItemV1"
+      discriminator:
+        propertyName: mealType
+        mapping:
+          SNACK: "#/components/schemas/RecipesCollectionDefaultItemV1"
+          LUNCH: "#/components/schemas/RecipesCollectionDefaultItemV1"
+          BREAKFAST: "#/components/schemas/RecipesCollectionDefaultItemV1"
+          DINNER: "#/components/schemas/RecipesCollectionDinnerItemV1"
+
+    RecipesCollectionBasicItemV1:
+      description: A single recipe.
+      type: object
+      required:
+        - id
+        - name
+        - ingredients
+        - preparationTime
+        - experienceLevel
+        - links
+      properties:
+        id:
+          description: The unique identifier of the recipe.
+          type: string
+        name:
+          description: The name of the recipe.
+          type: string
+        healthLabel:
+          description: Indicates how big your guilty conscience should be.
+          type: string
+        creationDate:
+          description: The date and time the recipe was added to the recipe collection.
+          type: string
+          format: date-time
+        ingredients:
+          description: Contains information about the ingredients required for the recipe.
+          type: array
+          items:
+            description: The list of ingredients.
+            required:
+              - name
+              - foodId
+              - quantity
+            properties:
+              name:
+                description: The ingredient.
+                type: string
+              foodId:
+                description: The unique identifier of the ingredient.
+                type: string
+              quantity:
+                description: The quantity of the ingredient required for four people.
+                type: object 
+                required:
+                  - value
+                  - unit
+                properties:
+                  value:
+                    description: The value related to the quantity. 
+                    type: integer
+                  unit:
+                    description: The unit related to the quantity.
+                    type: string
+        preparationTime:
+          description: The average preparation time in minutes.
+          type: integer
+        experienceLevel:
+          description: The recommended experience level for the recipe.
+          type: string
+          x-extensible-enum:
+            - value: BEGINNER
+              description: I can do deep frozen pizza. Maybe even fried eggs.
+            - value: ADVANCED
+              description: I can do rice and pasta. Sometimes even with sauce.
+            - value: PRO
+              description: I can do whatever you want!
+        _links:
```

Comment:
`application/hal+json` HATEOAS elements, although we only use `application/json`?